shared/src/iosMain/kotlin/co/touchlab/kampstarter/models/Platform.kt

```diff
@@ -0,0 +1,27 @@
+package co.touchlab.kampstarter.models
+
+import co.touchlab.stately.concurrency.GuardedStableRef
+import kotlinx.coroutines.*
+import platform.Foundation.NSThread
+import kotlin.coroutines.CoroutineContext
+
+@OptIn(InternalCoroutinesApi::class)
+internal actual fun CoroutineScope.childContext(): CoroutineContext {
+    val ktorJob = Job()
+
+    val ref = GuardedStableRef(ktorJob)
+
+    coroutineContext[Job]!!.invokeOnCompletion(onCancelling = true) { cancelCause ->
+        val parentJob = ref.state
+
+        if (cancelCause is CancellationException)
+            parentJob.cancel(cause = cancelCause)
+        else
+            parentJob.cancel()
+    }
+
+    return coroutineContext + Job() + coroutineContext[CoroutineExceptionHandler]!!
```

Comment:
The !! in all these implementations is ugly, and can be replaced with null checks if we want to make it safer to use on something that's not our `MainScope`